### PR TITLE
Adds error checking for ModalCreateFolder

### PR DESCRIPTION
**Pull request by webinista**
Closes department-of-veterans-affairs/messaging-fe#67
- Adds error check for:
  - Existing folder name.
  - Blank folder name field (including spaces-only folder names)
  - Disallowed characters.
- Disables submit button if there is an error.
- Adds maximum length config option, error messages for ModalCreateFolder.
- Changes top margin of va-modal-button-group. Refactoring va-modal SCSS.

### CODE REVIEW

File: src/js/messaging/reducers/folders.js [hunk 1]
```diff
@@ -62,6 +65,8 @@ export default function folders(state = initialState, action) {
     case SET_CURRENT_FOLDER:
       // The + forces +action.folderId to be a number
       return set('data.currentItem.persistFolder', +action.folderId, state);
+    case SET_FOLDER_NAME:
+      return set('data.newFolderName', action.folderName, state);
```
Comment (U-DON) [inline src/js/messaging/reducers/folders.js]: This should probably live in the modal reducer under `createNewFolder`. It is related with folders, but this state is more closely related with the `createNewFolder` modal itself.

File: src/js/messaging/reducers/folders.js [hunk 2]
```diff
@@ -62,6 +65,8 @@ export default function folders(state = initialState, action) {
     case SET_CURRENT_FOLDER:
       // The + forces +action.folderId to be a number
       return set('data.currentItem.persistFolder', +action.folderId, state);
+    case SET_FOLDER_NAME:
```
Comment (U-DON) [inline src/js/messaging/reducers/folders.js]: This should probably live in the modal reducer under `createNewFolder`. It is related with folders, but this state is more closely related with the `createNewFolder` modal itself.

I would also move the action creator to the modal action module.

**Review by U-DON — CHANGES_REQUESTED** [review]

File: src/js/messaging/containers/Main.jsx [hunk 3]
```diff
@@ -65,19 +77,21 @@ Main.propTypes = {
...
   };
 };
 
 const mapDispatchToProps = {
   toggleCreateFolderModal,
   toggleFolderNav,
   toggleManagedFolders,
-  setCurrentFolder
+  setCurrentFolder,
+  setFolderName
```
Comment (U-DON) [inline src/js/messaging/containers/Main.jsx]: Maybe call it `setNewFolderName`?

File: src/js/messaging/containers/Main.jsx [hunk 4]
```diff
@@ -18,13 +19,21 @@ class Main extends React.Component {
...
+    this.handleOnFolderNameChange = this.handleOnFolderNameChange.bind(this);
+    this.handleOnSubmit = this.handleOnSubmit.bind(this);
   }
 
   handleOnFolderChange(domEvent) {
     const folderId = domEvent.target.dataset.folderid;
     this.props.setCurrentFolder(folderId);
   }
 
+  handleOnFolderNameChange(field) {
```
Comment (U-DON) [inline src/js/messaging/containers/Main.jsx]: I typically see `handle*` and `on*` as complementary for naming event handling, but not together. Maybe simplify these to just `handle*`. For this one in particular, `handleNewFolderNameChange` to match the suggested change of action name to `setNewFolderName`.

File: src/sass/modules/_m-modal.scss [hunk 5]
```diff
@@ -44,24 +44,24 @@
...
-  top: 0;
-  
-  &:hover {
+  &-close {
     background-color: inherit;
-    color: $color-base;
-  }
-} 
+    color: rgba($color-base, .8);
+    font-size: 2.5rem;
```
Comment (U-DON) [inline src/sass/modules/_m-modal.scss]: The font-weight is displaying as normal for me and should be bold.

**Comment by U-DON** [comment]
Validation is working great. Only changes I'd like to see are name/namespace related.

File: src/sass/modules/_m-modal.scss (hunk 5 above)
Comment (webinista) [inline src/sass/modules/_m-modal.scss]: Punting this one over to Vanguard. It looks like every button with a `usa-button-` prefix needs to have a `span` inside in order for it to render as bold. Can fix in this project later, but this is a global issue that affects the entire site.

File: src/js/messaging/reducers/modals.js [hunk 6]
```diff
@@ -26,6 +29,8 @@ export default function modals(state = initialState, action) {
       return set('attachments.visible', !state.attachments.visible, state);
     case TOGGLE_CREATE_FOLDER:
       return set('createFolder.visible', !state.createFolder.visible, state);
+    case SET_FOLDER_NAME:
+      return set('createFolder.folderName', action.folderName, state);
```
Comment (U-DON) [inline src/js/messaging/reducers/modals.js]: Path also needs to be updated to `createFolder.newFolderName`. I guess also update `SET_FOLDER_NAME` -> `SET_NEW_FOLDER_NAME`.

**Comment by webinista** [comment]
Blocks work on #68.

**Comment by webinista** [comment]
Closing this. It's now a mess. Going to open a new one.

**Pull request closed without merging**